### PR TITLE
[PT-BR] TypeScript with React

### PR DESCRIPTION
**Pull request by helcioItiyama**
Fiquei em dúvida se teria que traduzir os nomes de funções e variáveis também...

@khaosdoctor, @alvarocamillont, @danilofuchs e @orta

### CODE REVIEW

**Comment by orta** [comment]
Nice one, I think this is the first portuguese translation in the playground, so tyou'll need to make a `pt/sections.json`  like this one: https://github.com/microsoft/TypeScript-Website/blob/v2/packages/playground-examples/copy/es/sections.json

**Comment by helcioItiyama** [comment]
@orta! Just did it :)

**Comment by orta** [comment]
'brigados! 

Just need someone to give an OK on the language and we're good to go - thanks!

File: packages/playground-examples/copy/pt/sections.json [hunk 1]
```diff
@@ -0,0 +1,54 @@
+{
+  "sections": [
+    {
+      "name": "JavaScript",
+      "id": "JavaScript",
+      "subtitle": "Veja como o TypeScript melhora o dia a dia com JavaScript com uma mínima sintaxe adicional."
```
Comment (danilofuchs) [inline packages/playground-examples/copy/pt/sections.json]:
```suggestion
      "subtitle": "Veja como o TypeScript melhora o dia a dia com JavaScript com o mínimo de sintaxe adicional."
```

File: packages/playground-examples/copy/pt/JavaScript/External APIs/TypeScript com React.tsx [hunk 2]
```diff
@@ -0,0 +1,128 @@
...
+
+//TypeScript suporta JSX, e providencia um amplo conjunto de ferramentas de 
+//tipagem que modela ricamente como os componentes se conectam.
+
+//Para entender como o TypeScript funciona com os componentes do React
+//podemos começar com exemplos genéricos:
+//
+
+//-exemplo: funções-genéricas
+//-exemplo: classes-genéricas
```
Comment (danilofuchs) [inline packages/playground-examples/copy/pt/JavaScript/External APIs/TypeScript com React.tsx]: This is a syntax for a hidden hyperlink.
See the example in EN. I think you may want to keep it the same.
```suggestion
// - example:generic-functions
// - example:generic-classes
```

File: packages/playground-examples/copy/pt/JavaScript/External APIs/TypeScript com React.tsx [hunk 3]
```diff
@@ -0,0 +1,128 @@
...
+
+import * as React from 'react';
+
+//Suas props são sua API pública, portanto vale a pena tomar tempo
+//utilizando JSDoc para explicar como funciona:
+
+export interface Props {
+  /** O nome do usuário */
+  name: string;
+  /** O nome deve ser renderizado em negrito */
```
Comment (danilofuchs) [inline packages/playground-examples/copy/pt/JavaScript/External APIs/TypeScript com React.tsx]:
```suggestion
  /** Renderiza o nome em negrito */
```

File: packages/playground-examples/copy/pt/JavaScript/External APIs/TypeScript com React.tsx [hunk 4]
```diff
@@ -0,0 +1,128 @@
...
+//{} em um atributo
+
+let username = "Cersei"
+const ShowStoredUser: React.FC<Props> = (props) => {
+  return <PrintName name={username} priority />
+}
+
+//TypeScript também funciona com código React moderno, aqui você
+//pode ver que count e setCount foram induzidos corretamente a usar números com base
+//no valor inicial passado para o useState.
```
Comment (danilofuchs) [inline packages/playground-examples/copy/pt/JavaScript/External APIs/TypeScript com React.tsx]:
```suggestion
//TypeScript também funciona com código React moderno. Aqui você
//pode ver que count e setCount foram induzidos corretamente a usar números
//com base no valor inicial passado para o useState.
```

**Comment by khaosdoctor** [comment]
@orta Does translating the file name implies in any problems? I just saw that the file name was translated to PT but I'm not sure what it implies to the project as a whole. I probably think it should stay in English.

**Comment by khaosdoctor** [comment]
From the language POV, I agree with the changes proposed by @danilofuchs :)

**Comment by orta** [comment]
Yeah, the filename definitely needs to stay the same, good spot @khaosdoctor  - I actually don't have a way to provide different display titles for the samples ATM, will have a think about how/if I can do that

**Comment by khaosdoctor** [comment]
@helcioItiyama Can you please revert the filename back to the English title?

**Comment by helcioItiyama** [comment]
@khaosdoctor done :)

**Review by khaosdoctor — APPROVED** [review]

**Comment by khaosdoctor** [comment]
@helcioItiyama I also think you're gonna need to apply @danilofuchs's changes by hand, if you haven't done that already

**Comment by orta** [comment]
haha, so much rebasing and fixes - thanks a lot @helcioItiyama !

**Comment by helcioItiyama** [comment]
@orta no worries hehe

**Comment by orta** [comment]
BTW, here it is: https://www.staging-typescript.org/pt/play/?jsx=2&esModuleInterop=true&q=334#example/typescript-with-react

**Comment by helcioItiyama** [comment]
Awesome! I'll do my best to keep helping!

**Pull request merged**